### PR TITLE
Fix bug in keys add events caching

### PR DESCRIPTION
**Pull request by Amuhar**

### CODE REVIEW

File: src/contracts/signing-key-events-cache/leveldb/leveldb.service.ts [hunk 1]
```diff
@@ -199,6 +205,28 @@ export class LevelDBService {
     await this.db.batch(ops);
   }
 
+  private validateHeader(
```
Comment (eddort) [inline src/contracts/signing-key-events-cache/leveldb/leveldb.service.ts:208]: Perhaps we should think about some kind of library for schema validation? Like zod?

File: src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts [hunk 2]
```diff
@@ -70,12 +76,18 @@ export class SigningKeyEventsCacheService {
...
     const firstNotCachedBlock = initialCache.headers.endBlock + 1;
     const toBlock = latestBlock - SIGNING_KEYS_EVENTS_CACHE_LAG_BLOCKS;
 
     const totalEventsCount = initialCache.data.length;
     let newEventsCount = 0;
 
+    const stakingModulesContracts =
+      await this.repositoryService.getCachedStakingModulesContracts();
+    const stakingModulesAddresses = Object.keys(stakingModulesContracts);
+
```
Comment (eddort) [inline src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts:90]: Should we replace this call with `getStakingModules`? As I see it, this function does the same job

Comment (eddort) [reply src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts:90]: And should we re-check the list of modules in each `getUpdatedSigningKeyEvents` call?

**Comment by eddort** [comment]
Great job! Thank you!

File: src/contracts/signing-key-events-cache/leveldb/leveldb.service.ts (hunk 1 above)
Comment (Amuhar) [reply src/contracts/signing-key-events-cache/leveldb/leveldb.service.ts:208]: I am not sure that we should install a library for just one check. If this check is enough, I will not delegate it to a library.

File: src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts (hunk 2 above)
Comment (Amuhar) [reply src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts:90]: I don't think so, as we check it in handleNewBlock and use getUpdatedSigningKeyEvents in the same iteration to search for duplicates.

**Review by eddort — APPROVED** [review]

**Pull request merged**